### PR TITLE
Upgrade to scala 2.13 and http4s 0.21.0

### PR DESCRIPTION
**Pull request by Igosuki**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -94,7 +91,7 @@ lazy val license = licenses in ThisBuild := Seq(
 
 lazy val buildSettings = publishing ++
   Seq(
-    scalaVersion := "2.12.8",
+    scalaVersion := "2.13.0",
     crossScalaVersions := Seq(scalaVersion.value, "2.11.12"),
```
Comment (lorandszakacs) [inline build.sbt]: `crossScalaVersions` also needs  `"2.12.8"`

File: build.sbt [hunk 2]
```diff
@@ -79,9 +78,7 @@ lazy val compileFlags = Seq(
   "-language:higherKinds",
   "-language:existentials",
   "-language:implicitConversions",
-  "-Ywarn-unused",
-  "-Ypartial-unification",
-  "-Xfatal-warnings"
+  "-Ywarn-unused"
```
Comment (lorandszakacs) [inline build.sbt]: Flags can be configured based on scala version. Something like this:
```
scalacOptions ++= (CrossVersion.partialVersion(scalaVersion.value) match {
        case Some((2, 11)) => scala2_11Flags
        case Some((2, 12)) => scala2_12Flags
        case Some((2, 13)) => scala2_13Flags
        case _             => Seq.empty
```

**Comment by Igosuki** [comment]
Yeah sorry I'll put out a compat folder like in circe :) coming back soon

**Comment by Igosuki** [comment]
Doesn't compile with 2.12 because some partial unification is failing on CompilerRoutes, trying to figure out why...

File: build.sbt [hunk 3]
```diff
@@ -95,8 +107,8 @@ lazy val license = licenses in ThisBuild := Seq(
 lazy val buildSettings = publishing ++
   Seq(
     scalaVersion := "2.12.8",
-    crossScalaVersions := Seq(scalaVersion.value, "2.11.12"),
-    scalacOptions ++= compileFlags,
+    crossScalaVersions := Seq(scalaVersion.value, "2.12.8", "2.11.12"),
```
Comment (lorandszakacs) [inline build.sbt]: you no longer have cross-compilation for version `2.13.0`, since the main `scalaVersion` is `2.12.8` again :smile:

Comment (Igosuki) [reply build.sbt]: Yeah don't worry I'm doing changes right now

Comment (Igosuki) [reply build.sbt]: It's back now, I need to find a way to make the part using MapView cross compile in rho-swagger (or refactor) and then I think it'll be over.

**Comment by Igosuki** [comment]
I fixed most issues, although now in 2.12, HttpRoutes doesn't have a SemigroupK, which blocks the examples module from compiling...

**Comment by Igosuki** [comment]
It looks like partial unification doesn't work properly :/

**Comment by Igosuki** [comment]
Hmm, it was a typo... shameful

**Comment by Igosuki** [comment]
I added placeholders that are missing from the current scala-collection-compat latest release and used in the demo code, I put them all in core, but maybe it will cause conflicts for other users down the line ?

File: build.sbt [hunk 4]
```diff
@@ -73,22 +85,32 @@ lazy val `rho-examples` = project
...
   "-Xfatal-warnings"
 )
 
+def versionSpecificEnabledFlags(version: String) = (CrossVersion.partialVersion(version) match {
+  case Some((2, 13)) => Seq.empty[String]
+  case _ => Seq("-Ypartial-unification")
+})
+
+def versionSpecificDisabledFlags(version: String) = (CrossVersion.partialVersion(version) match {
+  case Some((2, 13)) => Seq("-Xfatal-warnings")
```
Comment (zarthross) [inline build.sbt]: What is the reason for disabling fatal warnings for 2.13 ?

**Comment by lorandszakacs** [comment]
@Igosuki, anything blocking here? I could pick up the branch if you're too busy or something like that.

File: build.sbt (hunk 4 above)
Comment (RafalSumislawski) [reply build.sbt]: There's a false-positive "unreachable code" warning in `StaticContentService`. Cause most likely one of these bugs: https://github.com/scala/bug/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+unreachable+2.13

**Comment by Igosuki** [comment]
It was working fine on my laptop, but please go ahead. I hadn't had news in a while so I didn't touch it.

**Comment by zarthross** [comment]
@Igosuki Thank you for the initial work on this.  I thought this was still WIP since it had failing builds so I didn't give it a full review.  I'll try and be more proactive in the future.  I'm closing this in favor of #342

**Pull request closed without merging**